specification/inference/unified_inference/UnifiedRequest.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TaskType } from '@inference/_types/TaskType'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { RequestBase } from '@_types/Base'
+import { Id } from '@_types/common'
+import { Duration } from '@_types/Time'
+
+/**
+ * An object style representation of a single portion of a conversation.
+ */
+export interface ContentObject {
+  /**
+   * The text content.
+   */
+  text: string
+  /**
+   * The type of content.
+   */
+  type: string
+}
+
+/**
+ * The function that the model called.
+ */
+export interface ToolCallFunction {
+  /**
+   * The arguments to call the function with in JSON format.
+   */
+  arguments: string
+  /**
+   * The name of the function to call.
+   */
+  name: string
+}
+
+/**
+ * A tool call generated by the model.
+ */
+export interface ToolCall {
+  /**
+   * The identifier of the tool call.
+   */
+  id: string
+  /**
+   * The function that the model called.
+   */
+  function: ToolCallFunction
+  /**
+   * The type of the tool call.
+   */
+  type: string
+}
+
+/**
+ * An object representing part of the conversation.
+ */
+export interface Message {
+  /**
+   * The content of the message.
+   */
+  content: string | Array<ContentObject>
```

Comment:
Unions such as this one have to be handled in a separate type for them to be understandable by the static client, so here we need a new type:
```
/**
 * @codegen_names string, object
 */
export type MessageContent = string | Array<ContentObject>
```

```
content: MessageContent
```